webapp/src/errorList.tsx

```diff
@@ -158,20 +165,53 @@ class ErrorListItem extends React.Component<ErrorListItemProps, ErrorListItemSta
     render() {
         const {error, stackframe, location} = this.props
 
-        const message = stackframe ? lf("at {0} (line {1})", stackframe.funcInfo.functionName, location.line + 1)
-            : lf("Line {0}: {1}", error.endLine ? error.endLine + 1 : error.line + 1, error.messageText)
+        const message = stackframe ?
+            stackFrameMessageStringWithLineNumber(stackframe, location) :
+            errorMessageStringWithLineNumber(error.error);
+        const errorCount = stackframe ? 1 : error.count;
 
         return <div className={`item ${stackframe ? 'stackframe' : ''}`} role="button"
                     onClick={this.onErrorListItemClick}
                     onKeyDown={sui.fireClickOnEnter}
                     aria-label={lf("Go to {0}: {1}", stackframe ? '' : 'error', message)}
                     tabIndex={0}>
-            {message}
+            {message} {(errorCount <= 1) ? null : <div className="ui gray circular label countBubble">{errorCount}</div>}
         </div>
     }
 
     onErrorListItemClick() {
-        const location = this.props.stackframe ? this.props.location : this.props.error
+        const location = this.props.stackframe ? this.props.location : this.props.error.error
         this.props.revealError(location, this.props.index)
     }
 }
+
+function errorMessageStringWithLineNumber(error: pxtc.KsDiagnostic) {
+    return lf("Line {0}: {1}", error.endLine ? error.endLine + 1 : error.line + 1, error.messageText);
+}
+
+function stackFrameMessageStringWithLineNumber(stackframe: pxsim.StackFrameInfo, location: pxtc.LocationInfo) {
+    return lf("at {0} (line {1})", stackframe.funcInfo.functionName, location.line + 1);
+}
+
+function groupErrors(errors: pxtc.KsDiagnostic[]) {
+    const grouped: { [key: string]: GroupedError} = {};
```

Comment:
nit - we have a type def for maps: ``Map<GroupedError>``

Reply:
> nit - we have a type def for maps: `Map<GroupedError>`

Oh yes, the Map type!

Reply:
`{ [key: string]: GroupedError}` is standard TS syntax and we should prefer that IMO. I'd like to bring this up at the next code review day